src/agimus_pytroller.cpp

```diff
@@ -321,8 +368,15 @@ AgimusPytroller::update(const rclcpp::Time &time,
     std::copy(new_commands_.begin(), new_commands_.end(),
               new_commands_rt_.begin());
 
-    for (std::size_t i = 0; i < ordered_state_interfaces_.size(); i++) {
-      last_state_[i] = ordered_state_interfaces_[i].get().get_value();
+    auto &state = ordered_augmented_state_interfaces_;
+    for (std::size_t i = 0; i < state.size(); i++) {
+      if (std::holds_alternative<LoanedStateInterfaceRef>(state[i])) {
```

Comment:
why can't you simply let the polymorphism do the job?
I believe the CommandInterface and StateInterface inherit from the same class that should have the "get_value"
It would ease your pain. But I kinda like your approach it's fancy!